### PR TITLE
Formatted code with `black` and added GitHub Actions for linting/format checking/testing

### PR DESCRIPTION
**Pull request by xoen**
- [formatted code with `black`](https://github.com/ministryofjustice/analytics-platform-concourse-auth0-client-resource/pull/10/commits/13fb2dbcfc89dfbf63b035a70d545d694c4d73f6)
- [added GitHub Actions' workflow to lint using flake8, check formatting using `black` and run tests automatically](https://github.com/ministryofjustice/analytics-platform-concourse-auth0-client-resource/pull/10/commits/b72587b1de071bb9bf1749ff6597892950bb08a8)

### CODE REVIEW

File: .github/workflows/lint-check-formatting-and-test.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+name: Lint, check formatting and run tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (damacus) [inline .github/workflows/lint-check-formatting-and-test.yaml]:
```suggestion
```

This is possibly redundant and not wanted?

Comment (xoen) [reply .github/workflows/lint-check-formatting-and-test.yaml]: Is it? Ah OK, it looked like weird YAML but I'll change to:

```yaml
on:
  push:
    branches:
      - master
  pull_request:
```

Comment (xoen) [reply .github/workflows/lint-check-formatting-and-test.yaml]: Changed ✅

File: .github/workflows/lint-check-formatting-and-test.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Adapted from  GH documentation: https://docs.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Lint, check formatting and run tests
+
+on:
+  push:
+    branches:
+      - master
```
Comment (damacus) [inline .github/workflows/lint-check-formatting-and-test.yaml:8]: One, for the future, but we should move the main branch of this repo to main 👍

**Review by damacus — APPROVED** [review]

File: .github/workflows/lint-check-formatting-and-test.yaml (hunk 2 above)
Comment (xoen) [reply .github/workflows/lint-check-formatting-and-test.yaml:8]: Yeah, good point, I was tempted to do it but then I didn't. I'll tell you what, I'll merge this and I'll give it a try tomorrow.
AFAICT I just need to create a new `main` branch locally, then push it to `origin`, and then update the repository settings on GH and it should be basically it (and of course update the GH Action workflow)

**Pull request merged**